relationships.md

```diff
@@ -0,0 +1,324 @@
+        "leftType": {
+          "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/1"
+        },
+        "rightType": {
+          "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/2"
+        },
+        "self": {
+          "href": "https://dspace7-entities.atmire.com/rest/api/core/relationshiptypes/1"
+        }
+      },
+      "_embedded": {
+        "leftType": {
+          "id": 1,
+          "label": "Publication",
+          "type": "entitytype",
+          "_links": {
+            "self": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/1"
+            },
+            "relationshiptypes": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/1/relationshiptypes"
+            }
+          }
+        },
+        "rightType": {
+          "id": 2,
+          "label": "Person",
+          "type": "entitytype",
+          "_links": {
+            "self": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/2"
+            },
+            "relationshiptypes": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/2/relationshiptypes"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The [relationship type](relationshiptypes.md) is embedded
+The 2 items are included as HAL links but are not embedded
+
+## Creating a relationship
+
+**POST /api/core/relationships?relationshipType=<:relationshipType>**
+
+A new relationship between 2 items can be created by specifying both items in a uri-list body and specifying the relationshipType as a parameter
+
+A sample CURL command would be:
+```
+curl -i -X POST 'https://dspace7-entities.atmire.com/rest/api/core/relationships?relationshipType=1' -H 'Authorization: Bearer eyJhbGciO…' -H "Content-Type:text/uri-list" --data 'https://dspace7-entities.atmire.com/rest/api/core/items/12623672-25a9-4df2-ab36-699c4c240c7e \n https://dspace7-entities.atmire.com/rest/api/core/items/5a3f7c7a-d3df-419c-8a2-f00ede62c60a'
+```
+
+The uri-list should always contain exactly 2 items. The first item will be used as the left Item. The second item will be used as the right Item.
+The relationshipType parameter is mandatory as well
+
+## Updating a relationship
+
+**PUT /api/core/relationships/<:id>**
+
+Update the items in the relationship
+
+A sample CURL command would be:
+```
+curl -i -X PUT 'https://dspace7-entities.atmire.com/rest/api/core/relationships/891' -H 'Authorization: Bearer eyJhbGciO…' -H "Content-Type:text/uri-list" --data 'https://dspace7-entities.atmire.com/rest/api/core/items/12623672-25a9-4df2-ab36-699c4c240c7e \n https://dspace7-entities.atmire.com/rest/api/core/items/5a3f7c7a-d3df-419c-8a2-f00ede62c60a'
+```
+
+The uri-list should always contain exactly 2 items. The first item will be used as the left Item. The second item will be used as the right Item.
+The relationshipType is not modifiable
```

Comment:
This doesn't follow the usual rules. The PUT verb is used to update the information in the addressed resource, to update a linked resource a PUT request should be sent against
`/api/core/relationships/:id/leftItem` (or rightItem or relationshipType)

Please details which are the allowed changes and the expected return codes

Reply:
The design choice made during the entities meetings was to be consistent with the POST. This was decided by the group to be correct and consistent with https://github.com/DSpace/Rest7Contract/blob/master/README.md

The return codes have been included, the allowed changes were already described

Reply:
I don't find where the current contract state something that is consistent with the proposed behaviour.
The contract states that PUT update the target resource
https://github.com/DSpace/Rest7Contract/blob/master/README.md#on-single-resource-endpoints

and that PUT on the association subpath should be used in this case
https://github.com/DSpace/Rest7Contract/blob/master/README.md#on-sub-path-of-a-single-resource-endpoint-associations

Reply:
I've created https://jira.duraspace.org/browse/DS-4230 to discuss the change in the working group meetings.

Reply:
sorry but the current implementation has not yet approved and merged in the master. The method to update or create relationship are IMHO not in the scope of the current PR as the functionality to update/create entities is not yet in place. There is no need to rush on that, otherwise we will end with a first angular implementation based on a contract that has not yet reach a general agreement.
You can easily remove this change from the current PR and create a dedicated PR for the PUT verb to discuss. This is similar to what has been done in other PR when the changes were not strictly needed for the functionalities to implement and a wider discussion was demanded

Reply:
I've been giving a second though on this since our last meeting... and I agree with Andrea that, for this Preview Release, since we won't have the interface for managing relationships. Perhaps, if all agree, is better to temporally remove this part and have a further discussion about this after the Preview Release. 
Still, at this point, I'm not convinced that Andrea's proposal is the most suited for this case (based on `/api/core/items/<:uuid>/owningCollection`), firstly because I don't think we can compare them directly, the owning Collection is required for all Items and an Item might have, or not, relationships. Secondly, relationships and entities, as it was required, is a layer that repositories can ignore. Repositories can keep the same unaltered structure, without Entities. I think the base for this discussion should start from items, bundles and bitstreams relations case, not the owning collection case.

Reply:
All, I definitely see the points here (thanks!), and we should clean up this endpoint.  My main question though is whether removing this functionality (from the main code) *now* will take significant effort.  This might be a scenario were we need to consider making an exception just to move this feature along. (If you recall, we had to make a similar exception in the Submission & Workflow PRs, where features in the first PR were not really necessary until the second PR.  We made an exception because refactoring them out was more problem than it was worth.)

Keep in mind that just because this makes it to `master` doesn't mean it's *done* (or production ready). All it means is the feature is semi-stable, but we can highly prioritize a refactor of this endpoint to align with @abollini's recommendations for `PUT`.

Reply:
Hi @tdonohue ,
I understand your willingness to move the entities forward and please trust me, I'm open to make exception for minor things or formalities. That said, the request here are different from the situation that you cite about the MyDSpace. Just for the sake of completes the "exception" was only to perform the requested changes in a subsequent PR that was already existing at the time of the first PR review and already under review. This has saved time in doing the refactoring two times / resolving the conflict in the second PR.
I'm not asking about a future refactoring, the functionality here described and that is not needed by this PR, needs to be done in a different way. Additional functionalities out of scope for the MyDSpace PR were requested to be removed and the requests have been honored (some examples
https://github.com/DSpace/DSpace/pull/2391/commits/1252075ac9e90b72a75c0d12158f26818085d8af
https://github.com/DSpace/DSpace/pull/2391/commits/cc11f4955677eaedfae70c54eb08e940bb6067fe
https://github.com/DSpace/DSpace/pull/2391#discussion_r273433633 )

Again, my request should only imply the deletion of some line of code (the PUT implementation in the relationship controller), the related details here in the contract and reintroduce the jsonIgnore annotation for the place attribute. if I am mistaken and something else is required making this a significant effort please let me know and we can reconsider our options.

Reply:
@abollini : I can see your point here on `PUT`.  I've reached out to Lieven & @benbosman and it sounds like we can work to minimally comment out the current implementation of `PUT` (so that it can be reintroduced easily later once we finalize how it should behave) and remove/replace the description in this Contract with a "TODO" sort of message.

Reply:
@tdonohue the TODO has been included here now